src/app/shared/equation.ts

```diff
@@ -250,8 +249,24 @@ export class Equation {
       }
       return 'too complex';
     }
-    case TypeDiscriminator.Logarithmize:
+    case TypeDiscriminator.Log: {
+      const log = <Log>this.LHS;
+      if (!log.getAntilogarithm().contains(v)) {
+        const left = log.getBase();
+        const right = Equation.log(this.RHS, Equation.log(log.getAntilogarithm(), Equation.constantFromNumber(-1)));
+        return new Equation(left, right).solve(v);
+      }
       return 'too complex';
+    }
+    case TypeDiscriminator.Antilogarithmize: {
+      const antilogarithmize = <Antilogarithmize>this.LHS;
+      if (!antilogarithmize.getLogarithm().contains(v)) {
+        const left = antilogarithmize.getBase();
+        const right = Equation.antilog(this.RHS, Equation.antilog(antilogarithmize.getLogarithm(), Equation.constantFromNumber(-1)));
+        return new Equation(left, right).solve(v);
+      }
+      return 'too complex';
+    }
```

Comment:
Remove TypeDiscriminator.Antilogarithmize and the code that uses it. Instead, add a new static constructor to Equation which computes the 'n'-th root of a value 'x', by creating exp(x, exp(mul(n, constant(-1)))). Keep the code in dimension and ScalarAndDimension.